src/word_search_generator/__init__.py

```diff
@@ -76,39 +79,35 @@ def json(self) -> str:
         )
 
     @property
-    def level(self) -> int:
-        """The difficulty level of the puzzle."""
-        return self._level
+    def valid_directions(self) -> DirectionSet:
+        return self._valid_directions
 
-    @level.setter
-    def level(self, val: int):
-        """Set the difficulty level of the word search.
-
-        Level 1 (Easy): Words can go forward in directions
-        EAST (E), or SOUTH (S).
-        Puzzle size is small by default.
+    @valid_directions.setter
+    def valid_directions(self, val):
+        self._valid_directions = utils.validate_level(val)
+        self._reset_puzzle()
 
-        Level 2 (Intermediate): Words can go forward in directions
-        NORTHEAST (NE), EAST (E), SOUTHEAST (SE), or (S).
-        Puzzle size is medium by default.
+    def _set_level(self, lvl) -> None:
+        """Lvl should be an int, but putting that type hint in makes the linter upset"""
+        try:
+            self.valid_directions = lvl
```

Comment:
I know it's double work but to pass typing you could use `self.valid_directions = utils.validate_level(lvl)` here. This is really just a backport for the old "level" property so the double-work shouldn't be that big of a deal. I doubt many people were actually calling the getter for level anyway.